pkg/controller/deployment/deployment_controller.go

```diff
@@ -658,8 +658,8 @@ func (dc *DeploymentController) syncDeployment(ctx context.Context, key string)
 	// rollback is not re-entrant in case the underlying replica sets are updated with a new
 	// revision so we should ensure that we won't proceed to update replica sets until we
 	// make sure that the deployment has cleaned up its rollback spec in subsequent enqueues.
-	if getRollbackTo(d) != nil {
```

Comment:
IMO, the passed in result should not be used directly here.

Reply:
We have a lot of interaction logic with rs and dep in the `getAllReplicaSetsAndSyncRevision` method, and there are operations to update dep. I am not sure whether it will have an impact

Reply:
If I'm wrong, feel free to ignore